src/model/transaction/__tests__/insertFragmentIntoContentState-test.js

```diff
@@ -107,3 +118,277 @@ test('must apply multiblock fragments', () => {
+  );
+
+  assertInsertFragmentIntoContentState(
+    createContentBlockNodeFragment([
+      {
+        key: 'B',
+        text: '',
+        children: List(['C']),
+        nextSibling: 'E',
+      },
+      {
+        key: 'C',
+        parent: 'B',
+        text: '',
+        children: List(['D']),
+      },
+      {
+        key: 'D',
+        parent: 'C',
+        text: 'Delta',
+      },
+      {
+        key: 'E',
+        text: 'Delta',
+        prevSibling: 'B',
+      },
+    ]),
+    initialSelection,
+    initialContent,
+  );
+});
+
+test('must apply ContentBlockNode fragments merging first block when possible', () => {
+  const initialSelection = SelectionState.createEmpty('first');
+  const initialContent = contentState.set(
+    'blockMap',
+    createContentBlockNodeFragment([
+      {
+        key: 'first',
+        text: '',
+        nextSibling: 'second',
+      },
+      {
+        key: 'second',
+        text: '',
+        prevSibling: 'first',
+      },
+    ]),
+  );
+
+  assertInsertFragmentIntoContentState(
+    createContentBlockNodeFragment([
+      {
+        key: 'A',
+        text: 'Alpha',
+        nextSibling: 'B',
+      },
+      {
+        key: 'B',
+        text: '',
+        children: List(['C']),
+        prevSibling: 'A',
+      },
+      {
+        key: 'C',
+        parent: 'B',
+        text: '',
+        children: List(['D']),
+      },
+      {
+        key: 'D',
+        parent: 'C',
+        text: 'Delta',
+      },
+    ]),
+    initialSelection,
+    initialContent,
+  );
+});
+
+test('must apply ContentBlockNode fragments merging first block when possible along side nested blocks', () => {
+  const initialSelection = SelectionState.createEmpty('firstChild');
+  const initialContent = contentState.set(
+    'blockMap',
+    createContentBlockNodeFragment([
+      {
+        key: 'root',
+        text: '',
+        children: List(['firstChild', 'lastChild']),
+      },
+      {
+        key: 'firstChild',
+        parent: 'root',
+        text: '',
+        nextSibling: 'lastChild',
+      },
+      {
+        key: 'lastChild',
+        parent: 'root',
+        text: '',
+        prevSibling: 'firstChild',
+      },
+    ]),
+  );
+
+  assertInsertFragmentIntoContentState(
+    createContentBlockNodeFragment([
+      {
+        key: 'A',
+        text: 'Alpha',
+        nextSibling: 'B',
+      },
+      {
+        key: 'B',
+        text: '',
+        children: List(['C']),
+        prevSibling: 'A',
+        nextSibling: 'E',
+      },
+      {
+        key: 'C',
+        parent: 'B',
+        text: '',
+        children: List(['D']),
+      },
+      {
+        key: 'D',
+        parent: 'C',
+        text: 'Delta',
+      },
+      {
+        key: 'E',
+        text: 'Delta',
+        prevSibling: 'B',
+      },
+    ]),
+    initialSelection,
+    initialContent,
+  );
+});
+
+test('must throw an error when trying to apply ContentBlockNode fragments on target block that has children', () => {
```

Comment:
Lets rename this to something more meaningful like,

`must throw an error when trying to apply ContentBlockNode fragments when selection is on a block that has children`